### ISSUE TITLE
Need to be able tell the workload version of a charm by looking at charmhub

### ISSUE THREAD

**Issue by sed-i** [canonical/charmhub.io#1548]
Charms have the resource page, but it is opaque:
- it doesn't say which resource revision a given charm revision is linked to
- a resource revision doesn't say anything about the workload version

![image](https://user-images.githubusercontent.com/82407168/225730427-ccd8efd0-5711-4563-8606-955e8588861a.png)

Right now, we have to rely on CI or `charmcraft status` to know which belongs where:
```
$ charmcraft status prometheus-k8s
Track    Base                  Channel    Version    Revision    Resources               
latest   ubuntu 20.04 (amd64)  stable     103        103         prometheus-image (r83)  
                               candidate  103        103         prometheus-image (r83)  
                               beta       103        103         prometheus-image (r83)  
                               edge       113        113         prometheus-image (r92)  
1.0      ubuntu 20.04 (amd64)  stable     103        103         prometheus-image (r83)  
                               candidate  103        103         prometheus-image (r83)  
                               beta       103        103         prometheus-image (r83)  
                               edge       103        103         prometheus-image (r83)
```
and, armed with that knowledge, go and look into the history of `metadata.yaml`, hoping that `upstream-source` is in fact utilized:
```yaml
resources:
  prometheus-image:
    type: oci-image
    description: Container image for Prometheus
    upstream-source: ghcr.io/canonical/prometheus:latest
```
...only to find out that in this case it's `latest`.

Long story short: to know the workload version we currently have to deploy the charm first.

**Comment by rbarry82** [comment]
This is an easy fix simply by listing a digest/sha256 next to all revisions (if they're OCI images)
